examples/MicroserviceExample/WebApi/RabbitMqService.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Text;
+using Microsoft.Extensions.Logging;
+using OpenTelemetry.Context.Propagation;
+using RabbitMQ.Client;
+
+namespace WebApi
+{
+    public interface IRabbitMqService : IDisposable
+    {
+        string PublishMessage();
+    }
+
+    public class RabbitMqService : IRabbitMqService
+    {
+        private const string QueueName = "TestQueue";
+
+        private readonly ILogger<RabbitMqService> logger;
+        private readonly ConnectionFactory connectionFactory;
+        private readonly IConnection connection;
+        private readonly IModel channel;
+        private readonly ActivitySource activitySource;
+        private readonly ITextFormat textFormat;
+
+        public RabbitMqService(ILogger<RabbitMqService> logger)
+        {
+            this.logger = logger;
+
+            this.connectionFactory = new ConnectionFactory()
+            {
+                HostName = Environment.GetEnvironmentVariable("RABBIT_HOSTNAME") ?? "localhost",
+                UserName = "guest",
```

Comment:
Should we switch username & password to also use EnvVar if available? Just to be good internet citizens and encourage people to not hardcode their credentials.
```csharp
UserName = Environment.GetEnvironmentVariable("RABBIT_USERNAME") ?? "guest",`
Password = Environment.GetEnvironmentVariable("RABBIT_PASSWORD") ?? "guest",`
```

Reply:
👍 Good point.